### PR TITLE
Automatic file validation for components in src/pages

### PR DESCRIPTION
**Pull request by salmanmanekia**

### CODE REVIEW

**Comment by bodia-uz** [comment]
We got an issue with this PR.

With `src/pages/index.js` file like:
```js
import MdTemplate from '../templates/md';

// NOTE: use `MdTemplate` as index page template with docs/index.md markdown

export const pageQuery = graphql`
  query IndexPage {
    site {
      ...mdTemplateSiteFields
    }
    page: markdownRemark(fields: { slug: { eq: "/" } }) {
      ...mdTemplateMarkdownFields
    }
  }
`;

export default MdTemplate;
```
we have an error: 
```
...
You must import React at the top of the file for a React component to be valid
...
```

Fixed by `import React from 'react'` at the top of component, but its not required in our case.

**Comment by KyleAMathews** [comment]
Hmm good point — I guess we can only check for an export. Thanks for reporting the issue!

**Comment by dannywils** [comment]
This check also causes gatsby builds to fail when `babel-plugin-auto-import` is used to import react. For example, consider the following in .babelrc: 

```js
  "plugins": [
    [
      "auto-import", {
        "declarations": [
          { "default": "React", "path": "react" }
        ]
      }
    ]
  ]
```

If this babel plugin is now officially unsupported should it be added to the documentation?

**Comment by KyleAMathews** [comment]
@dannywils I just removed the check for `React` https://github.com/gatsbyjs/gatsby/pull/3949

We're now just checking if the file has a valid export.

File: packages/gatsby/src/redux/actions.js [hunk 1]
```diff
@@ -187,6 +187,72 @@ actions.createPage = (page: PageInput, plugin?: Plugin, traceId?: string) => {
...
+        console.log(`Add the following to the top of the component:`)
+        console.log(``)
+        console.log(`    import React from 'react'`)
+        console.log(``)
+      }
+
+      if (!includesDefaultExport) {
+        console.log(``)
+        console.log(
+          `The page component must export a React component for it to be valid`
```
Comment (jquense) [inline packages/gatsby/src/redux/actions.js]: These warnings are giving false positives on my site where the pages are `.md` files that are turned into components by webpack. I think this warning is a bit zealous, at the very least it probably should ignore none js files

Comment (riwsky) [reply packages/gatsby/src/redux/actions.js]: likewise, it fails to recognize
```
export {
 SomeRandomReactCompoment as default
}
```

would a PR adding "as default" to the above checks be acceptable?

Comment (DSchau) [reply packages/gatsby/src/redux/actions.js]: @riwsky sure - that seems reasonable!

**Comment by jlengstorf** [comment]
Hiya @salmanmanekia! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**